src/inmanta/ast/constraint/expression.py

```diff
@@ -219,8 +218,6 @@ def requires_emit(self, resolver: Resolver, queue: QueueScheduler) -> Dict[objec
         temp: ResultVariable = ResultVariable()
         temp.set_type(Bool())
 
-        temp.set_provider(self)
-
         # wait for the lhs
         HangUnit(queue, resolver, self.children[0].requires_emit(resolver, queue), temp, self)
```

Comment:
shouldn't the `HangUnit` also put a promise on temp, to replace the set_provider